compliance_checker/tests/test_ioos_profile.py

```diff
@@ -0,0 +1,293 @@
+        nc_obj = Dataset(os.devnull, 'w', diskless=True)
+        self.addCleanup(nc_obj.close)
+
+        # The dataset needs at least one variable to check that it's missing
+        # all the required attributes.
+        nc_obj.createDimension('time', 1)
+        nc_obj.createVariable('sample_var', 'd', ('time',))
+
+        sample_var = nc_obj.variables['sample_var']
+
+        results = self.ioos.check_variable_units(nc_obj)
+        self.assert_result_is_bad(results)
+
+        sample_var.units = 'm'
+        sample_var.short_name = 'sample_var'
+
+        results = self.ioos.check_variable_units(nc_obj)
+        self.assert_result_is_good(results)
+
+    def test_altitude_units(self):
+        '''
+        Tests that the altitude variable units test is working
+        '''
+
+        results = self.ioos.check_altitude_units(self.ds)
+        self.assert_result_is_good(results)
+
+        # Now test an nc file with a 'z' variable without units
+        # Create an empty dataset that writes to /dev/null This acts as a
+        # temporary netCDF file in-memory that never gets written to disk.
+        nc_obj = Dataset(os.devnull, 'w', diskless=True)
+        self.addCleanup(nc_obj.close)
+
+        # The dataset needs at least one variable to check that it's missing
+        # all the required attributes.
+        nc_obj.createDimension('time', 1)
+        nc_obj.createVariable('z', 'd', ('time',))
+        z = nc_obj.variables['z']
+        z.short_name = 'sample_var'
+
+        results = self.ioos.check_variable_units(nc_obj)
+        self.assert_result_is_bad(results)
+
+
+class TestIOOS1_1(BaseTestCase):
+    '''
+    Tests for the compliance checker implementation of IOOS Metadata Profile
+    for NetCDF, Version 1.1
+    '''
+
+    def setUp(self):
+        # Use the IOOS 1_1 dataset for testing
+        self.ds = self.load_dataset(STATIC_FILES['ioos_gold_1_1'])
+
+        self.ioos = IOOS1_1Check()
+
+    def test_cc_meta(self):
+        assert self.ioos._cc_spec == 'ioos'
+        assert self.ioos._cc_spec_version == '1.1'
+
+    def test_required_attributes(self):
+        '''
+        Tests that required attributes test is working properly
+        '''
+
+        results = self.ioos.check_high(self.ds)
+        for result in results:
+            self.assert_result_is_good(result)
+
+    def test_recomended_attributes(self):
+        '''
+        Tests that recommended attributes test is working properly
+        '''
+
+        results = self.ioos.check_recommended(self.ds)
+        for result in results:
+            self.assert_result_is_good(result)
+
+    def test_bad_platform_variables(self):
+        '''
+        Tests that the platform variable attributes check is working
+        '''
+
+        # Create an empty dataset that writes to /dev/null This acts as a
+        # temporary netCDF file in-memory that never gets written to disk.
+        nc_obj = Dataset(os.devnull, 'w', diskless=True)
```

Comment:
Keep in mind here that if you're doing some parallel test runs that multiple open files to `os.devnull` will throw an exception in the NetCDF library even though the file isn't actually going to be writing to disk.

Reply:
Good point, we use this all over the project. Can you suggest a better solution and create an issue to fix this for all the tests?